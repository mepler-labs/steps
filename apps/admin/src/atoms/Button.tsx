import React from 'react';
import styled from 'styled-components';

import {
  darken,
  green,
  white,
} from 'styles/colors';

interface Props {
  className?: string;
  children: any;
  white?: boolean;
}

class Button extends React.Component<Props, {}> {
  static defaultProps = { white: false };

  render() {
    const { className, children, white } = this.props;

    const ButtonEl = white ? WhiteButton : GreenButton;

    return <ButtonEl>{children}</ButtonEl>;
  }
}

const BaseButton = styled.button`
  border: none;
  border-radius: 1000px;
  cursor: pointer;
  display: inline-block;
  font-size: 0.825em;
  min-width: 180px;
  padding-bottom: 1.5em;
  padding-left: 2.25em;
  padding-right: 2.25em;
  padding-top: 1.5em;
  text-transform: uppercase;
`;

const WhiteButton = BaseButton.extend`
  background-color: ${white};
  color: ${green};
`;

const GreenButton = BaseButton.extend`
  background-color: ${green};
  color: ${white};

  &:hover {
    background-color: ${darken(green, 0.25)};
  }
`;

export default Button;
