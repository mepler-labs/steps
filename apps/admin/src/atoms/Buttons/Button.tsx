import React from 'react';
import styled from 'styled-components';

import { remCalc } from 'styles/type';
import { darken, green, white } from 'styles/colors';

interface Props {
  white?: boolean;
  type?: string;
  onClick?(event: any): void;
}

const Button: React.SFC<Props> = ({ white = false, children }) => {
  const ButtonEl = white ? WhiteButton : GreenButton;

  return <ButtonEl>{children}</ButtonEl>;
};

const BaseButton = styled.button`
  border: none;
  border-radius: 1000px;
  cursor: pointer;
  display: inline-block;
  font-size: 0.825em;
  min-width: 180px;
  padding-bottom: ${remCalc(20)};
  padding-left: ${remCalc(30)};
  padding-right: ${remCalc(30)};
  padding-top: ${remCalc(20)};
  text-transform: uppercase;
`;

const WhiteButton = BaseButton.extend`
  background-color: ${white};
  color: ${green};
`;

const GreenButton = BaseButton.extend`
  background-color: ${green};
  color: ${white};

  &:hover {
    background-color: ${darken(green, 0.25)};
  }
`;

export default Button;
