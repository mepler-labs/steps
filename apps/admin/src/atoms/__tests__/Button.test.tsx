import Button from '../Button';
import { shallow } from 'enzyme';
import * as React from 'react';
import 'jest';

describe('Button.tsx', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Button className="my-button-class">My button text</Button>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
