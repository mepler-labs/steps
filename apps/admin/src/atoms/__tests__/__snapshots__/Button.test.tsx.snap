// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.tsx renders correctly 1`] = `
<styled.button
  className="my-button-class"
>
  My button text
</styled.button>
`;
